enterprise/backend/src/metabase_enterprise/workspaces/api.clj

```diff
@@ -141,6 +143,34 @@
     {:transforms (for [transform transforms]
                    (assoc transform :workspace (get workspaces-by-id (tid->wid (:id transform)))))}))
 
+(api.macros/defendpoint :post "/:id/promote"
+  :- [:map
+      [:promoted [:sequential [:map [:id ms/PositiveInt] [:name :string]]]]
+      [:errors {:optional true} [:sequential [:map [:id ms/PositiveInt] [:name :string] [:error :string]]]]
+      [:workspace [:map [:id ms/PositiveInt] [:name :string]]]
+      [:archived-at :any]]
+  "Promote workspace transforms back to main Metabase and archive the workspace.
+
+  This will:
+  1. Update original transforms with workspace versions
+  2. Re-execute transforms in the original schema
+  3. Archive the workspace and clean up isolated resources
+
+  Returns a report of promoted transforms and any errors."
+  [{:keys [id]} :- [:map [:id ms/PositiveInt]]
+   _query-params
+   _body-params]
+  (let [ws               (u/prog1 (t2/select-one :model/Workspace :id id)
+                           (api/check-404 <>)
+                           (api/check-400 (nil? (:archived_at <>)) "Cannot promote an already archived workspace"))
+        {:keys [promoted
+                errors]} (ws.promotion/promote-transforms! ws)]
+    {:promoted    (vec promoted)
+     :errors      errors
+     :workspace   {:id id :name (:name ws)}
+     :archived_at (when-not errors
+                    (:archived-at (ws.cleanup/archive-workspace! id)))}))
```

Comment:
Note that promotion should not implicitly archive the workspace.

I think what you've implemented here would more accurately be called "merge".

Think about promotion as being a "make this live" button on a particular transform page for this goal.

Reply:
Do you want to drop that or make it optional?